lib/datadog/appsec/processor/rule_merger.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+module Datadog
+  module AppSec
+    class Processor
+      # RuleMerger merge different sources of information
+      # into the rules payload
+      module RuleMerger
+        class << self
+          def merge(rules:, data: nil, overrides: nil)
+            rules_dup = rules.dup
```

Comment:
Looking at the logic below it's a bit hard to track whether no nested mutation happens.

WDYT of adding some deep `freeze` to the `let`s in specs so that we ensure we never mutate the originals?